File: src/main/java/engineer/thomas_werner/euler/Problem9.java
package engineer.thomas_werner.euler;

public class Problem9 {

    public static void main(String[] args) {
        System.out.println(new Problem9().findPythagoreanTriplet());
    }

    long findPythagoreanTriplet() {
        for(int c=1; c<1_000; c++)
            for(int b=1; b<c; b++)
                for(int a=1; a<b; a++)
                    if(isPythagoreanTriplet(a,b,c) && isTargetTriplet(a,b,c))
                        return getResult(a,b,c);
        return 0;
    }

    boolean isPythagoreanTriplet(final int a, final int b, final int c) {
        return (a < b) && (b < c) && (a * a + b * b == c * c);
    }

    boolean isTargetTriplet(final int a, final int b, final int c) {
        return 1_000 == a + b + c;
    }

    long getResult(final int a, final int b, final int c) {
        System.out.println("a: " + a +", b: " + b +", c: " +c);
        return a * b * c;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem9.kt
package de.huddeldaddel.euler

/**
 * Solution for https://projecteuler.net/problem=9
 */
fun main() {
    println(Problem9().findPythagoreanTriplet())
}

class Problem9 {

    fun findPythagoreanTriplet(): Long {
        for (c in 1..999)
            for (b in 1 until c)
                for (a in 1 until b)
                    if (isPythagoreanTriplet(a, b, c) && isTargetTriplet(a, b, c))
                        return getResult(a, b, c)
        return 0
    }

    private fun isPythagoreanTriplet(a: Int, b: Int, c: Int): Boolean {
        return a < b && b < c && a * a + b * b == c * c
    }

    private fun isTargetTriplet(a: Int, b: Int, c: Int): Boolean {
        return 1000 == a + b + c
    }

    private fun getResult(a: Int, b: Int, c: Int): Long {
        println("a: $a, b: $b, c: $c")
        return (a * b * c).toLong()
    }

}
